op-challenger/game/fault/preimages/large.go

```diff
@@ -52,21 +52,37 @@ func NewLargePreimageUploader(logger log.Logger, txMgr txmgr.TxManager, contract
 	return &LargePreimageUploader{logger, txMgr, contract}
 }
 
+func NewContractLeaf(input []byte, index *big.Int, commitment common.Hash) contracts.Leaf {
+	// Pad the input to the 136 byte leaf size.
+	input = append(input, make([]byte, matrix.LeafSize-len(input))...)
```

Comment:
We shouldn't add this padding when posting the last leaf (ie when finalise is true) - the contract will add the padding for us using the correct keccak rules (and we save on call data costs).

I suspect `Input` should just be type `[]byte` instead of a fixed size array. Sorry I screwed that up earlier...

Reply:
Ok removed the block size restriction 👍